catalog/core/catalog-core-dynamic/src/main/java/ddf/catalog/data/dynamic/impl/DynamicMetacard.java

```diff
@@ -0,0 +1,644 @@
+ * <p/>
+ * <p>
+ * For what constitutes a compatible change in serialization, see <a href=
+ * "http://docs.oracle.com/javase/6/docs/platform/serialization/spec/version.html#6678" >Sun's
+ * Guidelines</a>.
+ * </p>
+ */
+public class DynamicMetacard implements Metacard, MetacardType {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DynamicMetacard.class);
+
+    // the base metacard for dynamic metacards
+    public static final String DYNAMIC = "dynamic";
+
+    // Constants provided by Metacard
+    //public static final String CREATED_DATE = "createdDate"; // -> CREATED
+    //public static final String MODIFIED_DATE = "modifiedDate"; // -> MODIFIED
+    //public static final String EXPIRATION_DATE = "expirationDate";  // -> EXPIRATION
+    //public static final String EFFECTIVE_DATE = "effectiveDate"; // -> EFFECTIVE
+    //public static final String ID = "id";
+    //public static final String THUMBNAIL = "thumbnail";
+    //public static final String TITLE = "title";
+    //public static final String METADATA = "metadata";
+    //public static final String POINT_OF_CONTACT = "pointOfContact";
+    //public static final String DESCRIPTION = "description";
+    //public static final String RESOURCE_URI = "resourceURI";
+    //public static final String RESOURCE_SIZE = "resourceSize";
+    //public static final String SECURITY = "security";
+
+    // Constants used in addition to those from Metacard
+    public static final String LOCATION = "location";
+
+    public static final String SOURCE_ID = "sourceId";
+
+    public static final String METACARD_TYPE = "metacardType";
+
+    public static final String CONTENT_TYPE_NAMESPACE = "contentTypeNamespace";
+
+    public static final String CONTENT_TYPE_NAME = "contentTypeName";
+
+    public static final String CONTENT_TYPE_VERSION = "contentTypeVersion";
+
+    private LazyDynaBean attributesBean;
+
+    public DynamicMetacard() {
+
+    }
+
+    /**
+     * Creates a {@link Metacard} from the provided {@link DynaClass} with empty
+     * {@link Attribute}s.
+     */
+    public DynamicMetacard(DynaClass dynaClass) {
+        attributesBean = new LazyDynaBean(dynaClass);
+    }
+
+    /**
+     * Creates a {@link Metacard} from the provided {@link LazyDynaBean} or another DynamicMetacard
+     * with empty {@link Attribute}s.
+     */
+    public DynamicMetacard(LazyDynaBean dynaBean) {
+        attributesBean = dynaBean;
+    }
+
+    public void setAttributesBean(LazyDynaBean dynaBean) {
+        attributesBean = dynaBean;
+    }
+
+    /**
+     * Returns the property of this DynamicMetacard as a {@link Attribute}.
+     *
+     * @param s name of the attribute whose value(s) are to be retrieved
+     * @return Attribute object containing the name and attributes for the specified property
+     */
+    @Override
+    public Attribute getAttribute(String s) {
+        List<Serializable> list = null;
+        Serializable obj = (Serializable) attributesBean.get(s);
```

Comment:
It looks like attributeBean can be null if the object was created via the default constructor.